File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) 
    {
        int max=p[0];
        int min=p[0];
        int f_max=0;
        int i;
        for(i=1;i<p.size();i++)
        {
            if(p[i]>max)
            {
                max=p[i];
            }
            else if(p[i]<min && i<p.size()-1)
            {
                max=p[i];
                min=p[i];
            }
            if(max-min>f_max)
            {
                f_max=max-min;
            }
        }
        return(f_max);
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int maxProfit = 0;
        int sum = 0;
        for(int i=0;i<prices.Length-1;i++)
        {
            if((sum + (prices[i+1] - prices[i])) > 0)
            {
                sum = sum + (prices[i+1] - prices[i]);
            }
            else if((prices[i+1] - prices[i]) < 0)
            {
                sum = 0;
            }
            else
            {
                sum = (prices[i+1] - prices[i]);
            }

            if(sum > maxProfit)
            {
                maxProfit = sum;
            }
        }
        return maxProfit;

        
    }
}